src/Components/Medicine/MedicineAdministrationSheet/index.tsx

```diff
@@ -28,37 +29,48 @@ const MedicineAdministrationSheet = ({ readonly, is_prn }: Props) => {
   const [showDiscontinued, setShowDiscontinued] = useState(false);
 
   const filters = { is_prn, prescription_type: "REGULAR", limit: 100 };
+  const [prescriptionList, setPrescriptionList] = useState<Prescription[]>([]);
 
   const { data, loading, refetch } = useQuery(
     MedicineRoutes.listPrescriptions,
     {
       pathParams: { consultation },
-      query: { ...filters, discontinued: showDiscontinued ? undefined : false },
+      query: { ...filters, discontinued: false },
     }
   );
 
   const discontinuedPrescriptions = useQuery(MedicineRoutes.listPrescriptions, {
     pathParams: { consultation },
     query: {
       ...filters,
-      limit: showDiscontinued ? 100 : 1,
+      limit: 100,
       discontinued: true,
     },
     prefetch: !showDiscontinued,
   });
 
   const discontinuedCount = discontinuedPrescriptions.data?.count;
 
+  useEffect(() => {
+    if (!showDiscontinued) {
+      setPrescriptionList(data?.results ?? []);
+    } else {
+      const regularList = data?.results ?? [];
+      const discontinuedList = discontinuedPrescriptions.data?.results ?? [];
+      setPrescriptionList([...regularList, ...discontinuedList]);
+    }
+  }, [data, showDiscontinued]);
```

Comment:
Why is a useEffect required for this?

We could simply do:
```suggestion
  const prescriptionsList = [...(data?.results ?? []), ...(discontinued.data?.results ?? []);
```

Reply:
should i replace it with useMemo instead? It works as you suggested but editor is showing warnings for dependencies

Reply:
@rithviknishad

Reply:
useMemo would be an unnecessary overhead. Prefer using `useMemo` only for heavier computes.

Reply:
Like this
```
const prescriptionList = useMemo(() => {
    if (!showDiscontinued) {
      return data?.results ?? [];
    } else {
      const regularList = data?.results ?? [];
      const discontinuedList = discontinuedPrescriptions.data?.results ?? [];
      return [...regularList, ...discontinuedList];
    }
  }, [
    data?.results,
    discontinuedPrescriptions.data?.results,
    showDiscontinued,
  ]);
```

Reply:
https://blog.logrocket.com/when-not-to-use-usememo-react-hook/